### PR TITLE
CRM457-1196: Application details in provider_updated state

### PR DESCRIPTION
**Pull request by patrick-laa**
## Description of change
 [Link to relevant ticket](https://dsdmoj.atlassian.net/browse/CRM457-1196)

In addition to the obvious stuff I:

- Added a way of downloading files you previously uploaded, as apparently we've never done that before
- Fixed a rubocop warning about a cop that's changed namespace
- Added an event ID to the new event we create, as we now need them for event deduplication in the app store.

### CODE REVIEW

File: app/presenters/prior_authority/check_answers/report.rb
Comment (jsugarman) [inline app/presenters/prior_authority/check_answers/report.rb:69]: I am using a method to extract logic

```
      def about_request_section
        [
          PrimaryQuoteCard.new(application, verbose: @verbose),
          AlternativeQuotesCard.new(application, verbose: @verbose),
          ReasonWhyCard.new(application),
          further_information_card,
        ].compact
      end

      private

      def further_information_card
        return unless application.further_informations.any?

        FurtherInformationCard.new(application)
      end
```


this could be extended with your logic after the fact fairly easily i imagine. I am just working on tests atm.

```
      def further_information_card
        return unless application.further_informations.any?
        
        if @verbose
           application.further_informations.map { CompactFurtherInformationCard.new(_1) }
        else
           FurtherInformationCard.new(application)
         end
      end
```

Comment (patrick-laa) [reply app/presenters/prior_authority/check_answers/report.rb:69]: Cool. I might need to wait until  #666 is merged and then rebase off that, otherwise my PR will just get in the way.

File: app/presenters/prior_authority/check_answers/compact_further_information_card.rb
Comment (dwhenry) [inline app/presenters/prior_authority/check_answers/compact_further_information_card.rb:44]:
```
parts = [response] + links.flat_map { [tag.br, _1] }
```

**Review by dwhenry — APPROVED** [review]

File: app/presenters/prior_authority/check_answers/compact_further_information_card.rb
Comment (patrick-laa) [reply app/presenters/prior_authority/check_answers/compact_further_information_card.rb:44]: THANK YOU! For the life of me I couldn't see how to make this less ugly.

**Pull request merged**
